pkg/ccl/backupccl/system_schema.go

```diff
@@ -69,7 +71,7 @@ type systemBackupConfiguration struct {
 	// migrationFunc performs the necessary migrations on the system table data in
 	// the crdb_temp staging table before it is loaded into the actual system
 	// table.
-	migrationFunc func(ctx context.Context, execCtx *sql.ExecutorConfig, txn *kv.Txn, tempTableName string) error
+	migrationFunc func(ctx context.Context, execCtx *sql.ExecutorConfig, txn *kv.Txn, tempTableName string, rekeys jobspb.DescRewriteMap) error
```

Comment:
Passing though, take it or leave it. It's definitely more code for the sake of making the convention harder to break. I think the below suggestion would be better than your comments.

1) what about making the migrationFunc field required and then having an explicit value that indicates that it's a no-op and fail on nil
2) make this function an interface (can have a function-based implementation) and have an implementation of that interface which is used to explicitly mark that there are no fields. In this form, you reject nil at init time and you'll call the method in the interface no matter what unless it's the sentinel saying there is nothing to migrate

Reply:
yeah, I started on merging optIn/optOut and migrationFunc into one field but backed it out for the sake of a more contained pr. I was thinking: a) rename to `restoreStrategy`, have a `const optOutOfRestore migrationFunc = nil` and then have a noop `func rawRowRestore = func (....) { return nil }` for the ones that have no func right now. But then I was looking at the fact that for each one that has a func, after we call it, we update the job (details, ugh) to reflect that has been called for resumption, so then switched `return nil` to `return errNoopMigtation` and checked that in the caller, but it was starting to get to be a lot of code that wasn't related to ID fixes contained here so I backed it out for a future diff.